filepath.cc

```diff
@@ -5,15 +5,15 @@
 namespace mold {
 
 std::string get_current_dir() {
-  std::string buf(8192, '\0');
+  std::string buf(PATH_MAX, '\0');
   getcwd(buf.data(), buf.size());
   buf.resize(buf.find_first_of('\0'));
   return buf;
 }
 
 std::string get_realpath(std::string_view path) {
-  char buf[8192];
-  if (!realpath(std::string(path).c_str(), buf))
```

Comment:
It looks like the new code always returns a string of PATH_MAX length filled with NUL bytes. I think just replacing 8192 with PATH_MAX is better.